azure/packages/test/scenario-runner/src/logger.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import crypto from "crypto";
+import fs from "fs";
+
+import { ITelemetryBaseEvent } from "@fluidframework/common-definitions";
+import { LazyPromise, assert } from "@fluidframework/common-utils";
+import { ChildLogger, TelemetryLogger } from "@fluidframework/telemetry-utils";
+import { ITelemetryBufferedLogger } from "@fluidframework/test-driver-definitions";
+
+import { pkgName, pkgVersion } from "./packageVersion";
+
+export interface LoggerConfig {
+    scenarioName?: string;
+    namespace?: string;
+    runId?: string;
+}
+
+class FileLogger extends TelemetryLogger implements ITelemetryBufferedLogger {
+    private error: boolean = false;
+    private readonly schema = new Map<string, number>();
+    private targetEvents: string[] = [];
+    private logs: ITelemetryBaseEvent[] = [];
+
+    public constructor(private readonly baseLogger?: ITelemetryBufferedLogger) {
+        super(undefined /* namespace */, { all: { testVersion: pkgVersion } });
+    }
+
+    public registerExpectedEvent(expectedEventNames: string[]) {
+        this.targetEvents = expectedEventNames;
+    }
+
+    async flush(runInfo?: { url: string; runId?: number }): Promise<void> {
+        // console.log("event----", this.targetEvents);
```

Comment:
Maybe remove this line if we intend to not include it in the final version? Or consider leaving it in but changing it to `console.debug()` if we want to keep it around for debugging purposes